### DISCUSSION TITLE
GDM + Wayland

### DISCUSSION THREAD

**Discussion by anasgets111** [rustdesk/rustdesk#9474] [General]
### Bug Description

Using Rustdesk to access my PC when not at home, as it was the only app that worked,
I setup my pc to only turn off the screen with Login screen when not used (no timed suspend or shutdown)
It **was working** fine with CachyOS + Gnome 46 + Wayland
When 47 came out, I **re-installed** the system to have a fresh gnome..
then Reskdesk started asking for which screen to share **every** time I use it from login screen,
not sure what changed but I will provide whatever logs needed to help fix this since it was working already.
I know Wayland is experimental, so would be nice contributing to fix it in this critical amazing app.

### How to Reproduce

- Setup Rustdesk with a password in a Gnome Linux PC
- access it from an android phone as a 1st time setup for screen selection and save it in Linux
- try accessing it again when screen is locked

### Expected Behavior

it should be able to show the login screen and I log in to my user normally

### Operating system(s) on local side and remote side

Android -> Linux

### RustDesk Version(s) on local side and remote side

1.3.1 -> 1.3.1

### Screenshots

![image](https://github.com/user-attachments/assets/1d6a5e2c-ca2b-4bac-ba4f-3acb798db356)
![Screenshot_20240926_111739](https://github.com/user-attachments/assets/45d31504-14db-445e-b647-a6539252f133)
this is when my PC is not locked

and when it's locked 
![Screenshot From 2024-09-26 11-23-34](https://github.com/user-attachments/assets/ee55fde9-69a7-43da-8554-397b2bc8cece)
https://github.com/user-attachments/assets/54324062-d740-47ba-a7c5-f6fdcf924993

### Additional Context

speculation: was GDM working in x.org in 46 and now it's Wayland even thought Shell was Wayland in both cases ?

**Comment by rustdesk** [comment]
> When 47 came out, I **re-installed** the system to have a fresh gnome..

This is out of our control.

**Comment by anasgets111** [comment]
-this is for people searching for something like this-
Found 2 workarounds,
first one was turning off auto screen-lock, 
so after the set time period my screen would turn off but not lock screen which wasn't a very good solution.

the second solution:
Turns out CachyOS has their mutter-cachyos package that replaces mutter for gnome,
which solved this issue for me (they have some fixes on top of official mutter)
and now I can login using Rustdesk.

![image](https://github.com/user-attachments/assets/5cc06fc7-7a2e-4a4b-be71-e0647b7d1c7a)
